client/go/outline/tun2socks/tunnel.go

```diff
@@ -39,6 +40,12 @@ type Tunnel interface {
 	UpdateUDPSupport() bool
 }
 
+// ConnectOutlineTunnelResult represents the result of [ConnectOutlineTunnel].
+type ConnectOutlineTunnelResult struct {
```

Comment:
I imagine we can't use Generics for a Result struct, right?

Reply:
Good suggestion, though Gemini told me gomobile doesn't support exporting generic types. I can try.

Reply:
As expected, `gobind` doesn't recognize generic types.